quiz-bundle-spec.md

```diff
@@ -0,0 +1,71 @@
+# Qwiklabs Quiz Bundle Specification
+
+**Version 1**
+
+> This is a DRAFT document. We welcome feedback as this format evolves.
+
+## `qwiklabs.yaml` Structure
+
+Here's a sample `qwiklabs.yaml` file with all nested details removed to make it easier to see the general file structure.
+
+```yml
+entity_type: Quiz
+schema_version: 1
+
+# The authoring title of the quiz - not visible to learners.
+title: Super Hard Quiz
+
+# The threshold grade that a student needs to achieve in order to count as "passing" the quiz.
+passing_percentage: 67
+
+# The "items" (i.e. questions) contained in this quiz.
+items: ...
+```
+
+Note that all of the localized content (stem, option titles, rationales, etc) are HTML content that may be displayed in various contexts. All of these chunks will be sanitized according to the restricted set in the [HTML spec](./html-spec.md).
+
+### Items
+Items are polymorphic - i.e. there are several different _item types_ that are defined slightly differently. `items` is an array of items defined correctly for their type as defined by the `type` attribute.
```

Comment:
For not wanting to confuse non-technical authors, the word "polymorphic" is a choice bit of CS jargon. :)

Check out the [Environment Resources](https://github.com/CloudVLab/qwiklabs-content-bundle-spec/blob/master/lab-bundle-spec.md#environment-resources) section of the lab bundle spec. It's far from perfect (reading it now is inspiring me to make some significant edits), but personally I find the general layout more clear.

- Why is this list of objects important?
- What do all the objects in this list have in common?
- What are the different variations of objects?

Reply:
@jgan isn't the main audience of this spec engineers who are building to the spec, since authors will normally author via GW?

Reply:
Correct. I was just poking fun at the fact that before that had been discussed, you were already using the word "polymorphic" :stuck_out_tongue_winking_eye: